bridge/ffi/src/c_api.rs

```diff
@@ -0,0 +1,216 @@
+use crate::{bindings::*, utils::*};
+use dsnp_graph_config::SchemaId;
+use dsnp_graph_core::{
+	api::api::{GraphAPI, GraphState},
+	dsnp::dsnp_types::DsnpUserId,
+};
+use std::{
+	mem::ManuallyDrop,
+	sync::{Arc, Mutex},
+};
+
+#[no_mangle]
+pub extern "C" fn print_hello_graph() {
+	println!("Hello, Graph!");
+}
+
+// Collection of GraphStates
+static mut GRAPH_STATES: Option<Arc<Mutex<Vec<*mut GraphState>>>> = None;
+
+// Initialize GraphState
+#[no_mangle]
+pub unsafe extern "C" fn initialize_graph_state(
+	environment: *const Environment,
+) -> *mut GraphState {
+	let environment = &*environment;
```

Comment:
We need to wrap all these api calls in `catch_unwind`. Even though we are not going to have any expected panics but since having a panic is an undefined state we should handle them
https://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/std/panic/fn.catch_unwind.html